WHY_DI.md

```diff
@@ -0,0 +1,27 @@
+# Why use dependency injection?
+
+This document describes the basics of the dependency injection pattern. What it is? And why it is a good pattern to apply to your app development?
```

Comment:
" What is it?"

nit: maybe drop the "And"

Reply:
I'll make them all semi-colon separated, into a single sentence.